public/app/plugins/datasource/cloudwatch/guards.ts

```diff
@@ -14,3 +15,7 @@ export const isCloudWatchAnnotationQuery = (
 
 export const isCloudWatchAnnotation = (query: unknown): query is AnnotationQuery<CloudWatchAnnotationQuery> =>
   (query as AnnotationQuery<CloudWatchAnnotationQuery>).target?.queryMode === 'Annotations';
+
+export const isCloudWatchQuery = (query: DataQuery): query is CloudWatchQuery => {
```

Comment:
The queries are currently sorted by datasource (see https://github.com/grafana/grafana/blob/91948ce5549a8aafec41e0ef474e4ce6bea071a3/public/app/features/dashboard/state/initDashboard.ts#L111) .

That made sense for the optional method in ds API implementation, it might not be the case now (with event bus solution)? 
Maybe we can remove the `getQueriesByDatasource` step in the `initDashboard` file?

Reply:
I think that as long as we use the event bus solution, we'll need to differentiate the queries based upon data source type. The `isCloudWatchQuery` type guard is needed because this PR changed typings of the event payload from `queries: Record<string, T[]>` to `queries: Record<string, DataQuery[]>`. Since then, we need to cast DataQueries to AzureMonitorQuery/CloudWatchQuery.

Reply:
Gotcha 👍 .
Since we filter it in the data sources, what do you think about removing the function sorting the queries by data sources in `initDashboard`? Or do you see a benefit to keeping it?

Reply:
Actually, the purpose of this filter was only to assert the correct typescript type, i.e from DataQuery to CloudWatchQuery. So I think we should keep the filtering in initDashboard to prevent other data sources from accidentally using queries that belong to other plugins. 

Actually @yaelleC, I think I should revert the changes made in common.ts in this PR. Even though I think it's more accurate, it becomes painfully hard to listen to these events if the consumer has to assert the type using typeguards. Think the long term solution is to move away from the event bus and use the data source api instead.